src/components/ha-service-picker.ts

```diff
@@ -114,11 +114,15 @@ class HaServicePicker extends LitElement {
       if (!filter) {
         return processedServices;
       }
-      return processedServices.filter(
-        (service) =>
-          service.service.toLowerCase().includes(filter) ||
-          service.name?.toLowerCase().includes(filter)
-      );
+      const split_filter = filter.split(" ");
+      return processedServices.filter((service) => {
+        const lower_service_name = service.name.toLowerCase();
+        const lower_service = service.service.toLowerCase();
+        return (
+          split_filter.every((f) => lower_service_name.includes(f)) ||
+          split_filter.every((f) => lower_service.includes(f))
```

Comment:
```suggestion
          split_filter.every((f) => lower_service_name.includes(f) || lower_service.includes(f))
```
can `split_filter` just be looped once, does this have an impact on results? 🤔

Reply:
Seems reasonable to do. It would slightly change the match behavior, but it doesn't seem wrong. 

As another idea we use the *Fuse* fuzzy search logic in a lot of places, I'm not certain if that makes sense here, but if you wanted a general less strict pattern matching for search that's another option that we've used in a few places.

Reply:
Or I guess for entities picker we have FuzzyFilterSort function, not sure if one or the other of those is better.

Reply:
`Fuse` is really cool, I think it's a bit too loose for this though. Looks really cool for something like products in ecom where you would always want to show something.

I tried with `FuzzyFilterSort` and it seemed too strict, only matching the exact term.

I just went with or'ing these two includes in the end. The only real change is being able to match on both `.` and `:` in the same filter (for English users). I guess other languages would be able to mix and match language.